src/common/dsp/SurgeVoice.cpp

```diff
@@ -200,8 +214,25 @@ void SurgeVoice::legato(int key, int velocity, char detune)
       state.portasrc_key = state.getPitch();
    else
    {
-      state.portasrc_key =
-          ((1 - state.portaphase) * state.portasrc_key + state.portaphase * state.getPitch());
+      // exponential or linear key traversal for the portamento
+      float phase;
+      switch (scene->portamento.porta_exp)
+      {
+      case -1:
```

Comment:
We could add an enum in SurgeStorage.h with names like porta_linear, porta_exp and port_log then use those rather than 0,1,-1 consistently

Reply:
Indeed - no magic numbers!